pkg/pillar/hypervisor/kvm.go

```diff
@@ -852,6 +882,27 @@ func (ctx KvmContext) CreateDomConfig(domainName string, config types.DomainConf
 			}
 		}
 	}
+	if len(canBusAssignments) != 0 {
+		canIfContext := struct {
+			Machine    string
+			IfName     string
+			HostIfName string
+			ID         int
+		}{Machine: ctx.devicemodel, IfName: "", HostIfName: "", ID: 0}
```

Comment:
You can just move this into the for-loop in line 895 then you don't need dummy values, then you can remove the if (line 885) which helps against having the code deeply nested.

Reply:
I agree reduce the scope of the variable, but I'm not sure if I want to change it, I just followed exactly how it's implemented for other I/O devices just on the code above this excerpt, like serial and PCI assignments... there will be performance improvements? Otherwise I prefer to keep the same pattern for better code readability...

Reply:
resolved